pyston/test/test_recursion_limit.py

```diff
@@ -0,0 +1,43 @@
+import sys
+import threading
+import time
+
+def measure():
+    "Measures the current stack limit of python frames"
+    x = [0]
+
+    def inner():
+        x[0] += 1
+        inner()
+
+    try:
+        inner()
+    except RecursionError:
+        pass
+
+    return x[0]
+
+if __name__ == "__main__":
+    set_limit = 1024
+
+    # Basic test: check that setting the limit is close
+    sys.setrecursionlimit(set_limit)
+    actual_limit = measure()
+    assert set_limit - 3 <= actual_limit < 2 * set_limit, actual_limit
+
+    # Test that doubling the limit roughly doubles the number of frames
+    sys.setrecursionlimit(set_limit * 2)
+    actual_limit2 = measure()
+    assert 2 * actual_limit - 1 <= actual_limit2 <= 2.1 * actual_limit + 2, (actual_limit, actual_limit2)
```

Comment:
Ok I have just noticed that depending on the value of  `JIT_MIN_RUNS` this assert fails. I guess it's a bit too specific `assert 1.9 * acutal_limit` makes it work for me an all thresholds I tried